spec/unit/provider/package/yum_spec.rb

```diff
@@ -153,14 +153,14 @@
         allow(@yum_cache).to receive(:yum_binary=).with("yum")
         @provider = Chef::Provider::Package::Yum.new(@new_resource, @run_context)
         @provider.load_current_resource
-        expect(@provider.new_resource.package_name).to eq("testing")
+        expect(@provider.new_resource.package_name).to eq("testing.noarch")
```

Comment:
From my viewpoint specifying the architecture for packages is really a mess.  It might be harder to fix than I have time for.  I'm working on adding tests for some more cases.  What I did above was add the architecture to the package name so that the arch was preserved after going back through package.rb method target_version_array.   If the resource package name needs to be preserved as originally specified I'll have to figure out some other way of associating the package and the architecture.   I think documenting the use cases won't be a waste at any rate.

Thanks,
  Mark